pkg/proxy/handler.go

```diff
@@ -56,7 +58,24 @@ func (h *Handler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	defer complete()
 
 	log.Println("Waiting for IPs", id)
-	host := h.Endpoints.GetHost(r.Context(), deploy, "http")
+	host, err := h.Endpoints.AwaitHostAddress(r.Context(), deploy, "http")
+	if err != nil {
+		log.Printf("error while finding the host address %v", err)
+		switch {
+		case errors.Is(err, context.Canceled):
+			w.WriteHeader(http.StatusInternalServerError)
```

Comment:
I did some research on the inet what http code makes sense but looks like this case is not handled explicit very often. Alternatively [`499`](https://http.dev/499) was suggested, which is not part of Go stdlib though